packages/server/src/mL/Trainer.ts

```diff
@@ -184,8 +184,8 @@ export default class Trainer {
     const METRICSPATH = path.posix.join(this.project.directory, "metrics.json");
     if (fs.existsSync(METRICSPATH)) {
       const metrics = JSON.parse(fs.readFileSync(METRICSPATH, "utf8"));
-      this.epoch = Object.keys(metrics.precision).length;
       const jssteps: number[] = Object.keys(metrics.precision).map((s) => parseInt(s)); //we must change the metrics.json soon
+      this.epoch = jssteps.sort()[jssteps.length - 1];
```

Comment:
```suggestion
      this.epoch = jssteps.length > 0 ? jssteps.sort()[jssteps.length - 1] : 0;
```
We may want to do this just so any future changes to the behavior of the metrics.json don't break anything.